Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/DriverPool/Config.hs

```diff
@@ -14,69 +14,35 @@
 
 module SharedLogic.DriverPool.Config where
 
+import Domain.Types.Configs.DriverPoolConfig
+import Domain.Types.Merchant
 import Kernel.Prelude
 import Kernel.Types.Common
-import qualified Kernel.Types.SlidingWindowCounters as SWC
-import Kernel.Utils.Dhall (FromDhall)
-
-data IntelligentPoolConfig = IntelligentPoolConfig
-  { minQuotesToQualifyForIntelligentPool :: Int,
-    minQuotesToQualifyForIntelligentPoolWindowOption :: SWC.SlidingWindowOptions
-  }
-  deriving (Generic, FromDhall)
+import Kernel.Types.Error
+import Kernel.Types.Id
+import Kernel.Utils.Error
+import Storage.CachedQueries.CacheConfig (CacheFlow)
+import qualified Storage.CachedQueries.Configs.DriverPoolConfig as CDP
 
 data CancellationScoreRelatedConfig = CancellationScoreRelatedConfig
   { popupDelayToAddAsPenalty :: Maybe Seconds,
     thresholdCancellationScore :: Maybe Int,
     thresholdRidesCount :: Maybe Int
   }
-  deriving (Generic, FromDhall)
-
-data DriverPoolConfig = DriverPoolConfig
-  { minRadiusOfSearch :: Meters,
-    maxRadiusOfSearch :: Meters,
-    radiusStepSize :: Meters,
-    driverPositionInfoExpiry :: Maybe Seconds,
-    actualDistanceThreshold :: Maybe Meters,
-    maxDriverQuotesRequired :: Int,
-    driverQuoteLimit :: Int,
-    intelligentPoolPercentage :: Maybe Int,
-    driverRequestCountLimit :: Int
-  }
-  deriving (Generic, FromDhall)
-
-data OverrideDriverPoolConfigRange = OverrideDriverPoolConfigRange
-  { startDistance :: Meters,
-    endDistance :: Maybe Meters
-  }
-  deriving (Generic, FromDhall)
-
-data OverrideDriverPoolConfig = OverrideDriverPoolConfig
-  { configRange :: OverrideDriverPoolConfigRange,
-    driverPoolCfg :: DriverPoolConfig
-  }
-  deriving (Generic, FromDhall)
-
-type HasDriverPoolConfig r =
-  ( HasField "driverPoolCfg" r DriverPoolConfig,
-    HasField "overrideDriverPoolConfig" r [OverrideDriverPoolConfig],
-    HasField "intelligentPoolConfig" r IntelligentPoolConfig,
-    HasField "cancellationScoreRelatedConfig" r CancellationScoreRelatedConfig,
-    HasField "defaultPopupDelay" r Seconds,
-    HasMaxParallelSearchRequests r
-  )
+  deriving (Generic)
 
 type HasMaxParallelSearchRequests r =
   ( HasField "maxParallelSearchRequests" r Int
```

Comment:
This field should be a part of `SendSearchRequestJobConfig`

Reply:
Yeah. @0utkarsh Move this too.